app/controllers/listings_controller.rb

```diff
@@ -179,8 +183,10 @@ def create
     end
 
     params[:listing] = normalize_price_param(params[:listing]);
+    shape = get_shape(Maybe(params)[:listing][:transaction_type_id].to_i.or_else(nil))
+    unit_type = Maybe(shape[:units].first)[:type].or_else(nil)
```

Comment:
So shape can define multiple unit types but we always choose just the first one? What's the logic in this, why do we need :units to be an array?